packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -478,7 +446,27 @@ Future<int> buildAndroid(
     return 1;
   }
 
-  if (!force && !_needsRebuild(outputFile, manifest)) {
+  Map<String, File> extraFiles = <String, File>{};
+  for (String addFile in addFiles ?? <String>[]) {
```

Comment:
can we parse the --addFiles flag in the caller and pass the additional files as a map from APK paths to local files?